trk/src/Components/UserEdit.js

```diff
@@ -0,0 +1,206 @@
+import React, { useState, useContext, useEffect, useRef } from "react";
+import { View, Text, Button, TextInput, StyleSheet, Alert, TouchableOpacity, ScrollView, SafeAreaView, Dimensions } from "react-native";
+import firestore from '@react-native-firebase/firestore';
+import { AuthContext } from "../Utils/AuthContext";
+import { useNavigation } from '@react-navigation/native';
+import storage from '@react-native-firebase/storage';
+import ListItemSessions from "./ListItemSessions";
+import { Image } from "react-native";
+import ClimbItem from "./ClimbItem";
+import ListHistory from "./ListHistory";
+import { FlatList } from "react-native-gesture-handler";
+import { ActivityIndicator } from "react-native-paper";
+import {Modal} from "react-native-paper";
+import Icon from 'react-native-vector-icons/MaterialIcons'; // Or any other icon family you prefer
+import TapsApi from "../api/TapsApi";
+import ImagePicker from 'react-native-image-crop-picker';
+import { Line, Svg } from "react-native-svg";
+import UsersApi from "../api/UsersApi";
+
+const UserEdit = ({route}) => {
+    const navigation = useNavigation();
+    //let data = route.params.data;
+    //const title = route.params.title;
+    const {currentUser, role} = useContext(AuthContext);
+    const [user, setUser] = useState(route.params.user);
+
+    const {getUsersBySomeField} = UsersApi();
```

Comment:
Invoking database functions outside of useEffect hooks tends to generate excessive db calls. That's because our API files (`UsersApi` is an example) call the db whenever they're invoked, which could be every time the component renders. I haven't tested your code and so can't assess how often `UserEdit` is rendered when a user interacts with the component. But.. if it's a lot, it results in a loooot of db calls.

I'd advise to put a `console.log("[TEST] UserEdit called")` statement here (see above), to test it yourself.